orchard-ws/app/utils/AuthorizationSettings.scala

```diff
@@ -21,13 +21,12 @@ class AuthorizationSettings private (config: Config) {
     .entrySet()
     .asScala
     .filter(_.getKey.startsWith(s"hashed-keys."))
-    .map(kv => {
-      val key: String = kv.getKey.stripPrefix(s"hashed-keys.")
+    .foldLeft(Map[String, List[String]]()) { (b, kv) =>
+      val k: String = kv.getKey.stripPrefix(s"hashed-keys.")
       val configList: ConfigList = kv.getValue.asInstanceOf[ConfigList]
-      val listValue = configList.unwrapped().asScala.map(_.toString).toList
-      (key, listValue)
-    })
-    .toMap
+      val v = configList.unwrapped().asScala.map(_.toString).toList
+      (b -- v) ++ v.map(j => (j, b.getOrElse(j, List.empty) ++ List(k))).toMap
+    }
```

Comment:
also this method would be a lot more readable if we implement it like this:

```
val userRoles = for {
  (r, us) <- config.getConfig("hashed-keys").root().asScala.toSeq
  u <- asInstanceOf[ConfigList].unwrapped().asScala
} yield u -> r

userRoles.groupBy(_._1).view.mapValues(_.map(_._2)).toMap
```

Reply:
@realstraw looks good, let me change

Reply:
@realstraw changed